src/components/views/messages/DecryptionFailureBody.tsx

```diff
@@ -23,8 +23,10 @@ import { IBodyProps } from "./IBodyProps";
 import { LocalDeviceVerificationStateContext } from "../../../contexts/LocalDeviceVerificationStateContext";
 
 function getErrorMessage(mxEvent: MatrixEvent, isVerified: boolean | undefined): string {
-    if (mxEvent.isEncryptedDisabledForUnverifiedDevices) return _t("timeline|decryption_failure|blocked");
     switch (mxEvent.decryptionFailureReason) {
+        case DecryptionFailureCode.MEGOLM_KEY_WITHHELD_FOR_UNVERIFIED_DEVICE:
+            return _t("timeline|decryption_failure|blocked");
```

Comment:
This would say `The sender has blocked you from receiving this message`. Is there a reason we can't append `because your device is unverified` to the message?

Reply:
great question. I've avoided getting into changing the UI as part of this PR, and stuck to the posthog changes, but I'll raise this with the crypto team

Reply:
https://github.com/matrix-org/matrix-react-sdk/pull/12761